src/v/cloud_storage/topic_manifest.cc

```diff
@@ -353,53 +353,94 @@ void topic_manifest::do_update(const topic_manifest_handler& handler) {
-        topic_manifest::do_update(handler);
-    } else {
-        rapidjson::ParseErrorCode e = reader.GetParseErrorCode();
-        size_t o = reader.GetErrorOffset();
 
-        if (_topic_config) {
-            throw std::runtime_error(fmt_with_ctx(
-              fmt::format,
-              "Failed to parse topic manifest {}: {} at offset {}",
-              get_manifest_path(),
-              rapidjson::GetParseError_En(e),
-              o));
+    vlog(cst_log.debug, "Parsing topic manifest with format {}", format);
+
+    switch (format) {
+    case manifest_format::json: {
+        iobuf_istreambuf ibuf(result);
+        std::istream stream(&ibuf);
+
+        json::IStreamWrapper wrapper(stream);
+        json::Reader reader;
+        topic_manifest_handler handler;
+        if (reader.Parse(wrapper, handler)) {
+            vlog(cst_log.debug, "Parsed successfully!");
+            topic_manifest::do_update(handler);
         } else {
-            throw std::runtime_error(fmt_with_ctx(
-              fmt::format,
-              "Failed to parse topic manifest: {} at offset {}",
-              rapidjson::GetParseError_En(e),
-              o));
+            rapidjson::ParseErrorCode e = reader.GetParseErrorCode();
+            size_t o = reader.GetErrorOffset();
+
+            if (_topic_config) {
+                throw std::runtime_error(fmt_with_ctx(
+                  fmt::format,
+                  "Failed to parse topic manifest {}: {} at offset {}",
+                  get_manifest_path(),
+                  rapidjson::GetParseError_En(e),
+                  o));
+            } else {
+                throw std::runtime_error(fmt_with_ctx(
+                  fmt::format,
+                  "Failed to parse topic manifest: {} at offset {}",
+                  rapidjson::GetParseError_En(e),
+                  o));
+            }
         }
+        break;
+    }
+    case manifest_format::serde:
+        // serde format is straightforward: the buffer is a
+        // cluster::topic_configuration and _rev is inside
+        // remote_topic_properties
+        _topic_config = serde::from_iobuf<cluster::topic_configuration>(
+          std::move(result));
+        _rev = _topic_config->properties.remote_topic_properties.value()
+                 .remote_revision;
+        _manifest_version = topic_manifest::serde_version;
+        break;
     }
+
     co_return;
 }
 
 ss::future<serialized_data_stream> topic_manifest::serialize() const {
-    iobuf serialized;
-    iobuf_ostreambuf obuf(serialized);
-    std::ostream os(&obuf);
-    serialize(os);
-    if (!os.good()) {
-        throw std::runtime_error(fmt_with_ctx(
-          fmt::format,
-          "could not serialize topic manifest {}",
-          get_manifest_path()));
+    if (!_topic_config.has_value()) {
+        ss::throw_with_backtrace<std::runtime_error>(
+          "_topic_config is not initialized");
+    }
+
+    if (_manifest_version == first_version) {
```

Comment:
mixed-version cluster. the newer node wants to create a topic_manifest, so it will use json. 

the alternative is to pause progress while in this state, like we did for the partition_manifest json->serde transition. but that too was controversial